src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/AutoInsert/PreferHtmlInAttributeValuesDocumentPositionStrategy.cs

```diff
@@ -28,22 +27,20 @@ internal class PreferHtmlInAttributeValuesDocumentPositionInfoStrategy : IDocume
             return null;
         }
 
+        var absolutePosition = defaultDocumentPositionInfo.HostDocumentIndex;
         if (defaultDocumentPositionInfo.LanguageKind != RazorLanguageKind.Razor ||
-            defaultDocumentPositionInfo.HostDocumentIndex < 1)
-        {
-            return defaultDocumentPositionInfo;
-        }
-
-        var codeDocument = await documentContext.GetCodeDocumentAsync(cancellationToken).ConfigureAwait(false);
-        var previousPosition = defaultDocumentPositionInfo.HostDocumentIndex - 1;
-        var charBeforePosition = codeDocument.GetSourceText()[previousPosition];
-        if (charBeforePosition != '=')
+            absolutePosition < 1)
         {
             return defaultDocumentPositionInfo;
         }
 
+        // Get the node at previous position to see if we are after markup tag helper attribute,
+        // and more specifically after the EqualsToken of it
+        var previousPosition = absolutePosition - 1;
         var owner = await documentContext.GetSyntaxNodeAsync(previousPosition, cancellationToken).ConfigureAwait(false);
-        if (owner is null || owner.Kind != SyntaxKind.MarkupTagHelperAttribute)
+        if (owner is not MarkupTagHelperAttributeSyntax markupTagHelperAttributeSyntax
+            || markupTagHelperAttributeSyntax.EqualsToken.Width != 1
+            || markupTagHelperAttributeSyntax.EqualsToken.EndPosition != defaultDocumentPositionInfo.HostDocumentIndex)
```

Comment:
Nit: I'd find this easier to read if it was inverted to be the positive case, and since there isn't much code after the if, making it a guard doesn't matter too much I don't think. Also could use patters a bit (I'm assing the Width check here is to see if the equals sign is present?), eg:

```
if (owner is MarkupTagHelperAttributeSyntax { EqualsToken: { IsMissing: false } equalsToken } &&
    equalsToken.EndPosition == defaultDocumentPositionInfo.HostDocumentIndex)
{
  return good stuff;
}

return default case;

Reply:
agree. Pattern matching will help alot here :)

Reply:
I like that, added.